### PR TITLE
Add single checkbox helper

### PR DESCRIPTION
**Pull request by asprouc**

### CODE REVIEW

File: GovUkDesignSystem/GovUkHtmlHelperExtensions.cs [hunk 1]
```diff
@@ -77,6 +77,13 @@ public static class GovUkHtmlHelperExtensions
             return await htmlHelper.PartialAsync("/GovUkDesignSystemComponents/Checkboxes.cshtml", checkboxesViewModel);
         }
 
+        //public static async Task<IHtmlContent> GovUkCheckbox(
```
Comment (DanCorder) [inline GovUkDesignSystem/GovUkHtmlHelperExtensions.cs]: Please delete

File: GovUkDesignSystem/GovUkDesignSystemComponents/CheckboxViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
+﻿using System.Collections.Generic;
```
Comment (DanCorder) [inline GovUkDesignSystem/GovUkDesignSystemComponents/CheckboxViewModel.cs]: This file isn't used?

File: GovUkDesignSystem/GovUkDesignSystemComponents/Checkbox.cshtml [hunk 3]
```diff
@@ -0,0 +1,65 @@
+﻿@*@using GovUkDesignSystem.GovUkDesignSystemComponents
```
Comment (DanCorder) [inline GovUkDesignSystem/GovUkDesignSystemComponents/Checkbox.cshtml]: This file isn't used?

**Review by DanCorder — CHANGES_REQUESTED** [review]

File: GovUkDesignSystem/GovUkDesignSystemComponents/CheckboxViewModel.cs (hunk 2 above)
Comment (DanCorder) [reply GovUkDesignSystem/GovUkDesignSystemComponents/CheckboxViewModel.cs]: This file is still here

**Review by DanCorder — CHANGES_REQUESTED** [review]

File: GovUkDesignSystem/GovUkDesignSystemComponents/Checkbox.cshtml (hunk 3 above)
Comment (asprouc) [reply GovUkDesignSystem/GovUkDesignSystemComponents/Checkbox.cshtml]: Removed

File: GovUkDesignSystem/GovUkDesignSystemComponents/CheckboxViewModel.cs (hunk 2 above)
Comment (asprouc) [reply GovUkDesignSystem/GovUkDesignSystemComponents/CheckboxViewModel.cs]: Removed

File: GovUkDesignSystem/GovUkHtmlHelperExtensions.cs (hunk 1 above)
Comment (asprouc) [reply GovUkDesignSystem/GovUkHtmlHelperExtensions.cs]: Deleted

File: GovUkDesignSystem/GovUkHtmlHelperExtensions.cs [hunk 4]
```diff
@@ -124,6 +124,29 @@ public static class GovUkHtmlHelperExtensions
                 idPrefix);
         }
 
+        public static async Task<IHtmlContent> GovUkCheckboxFromBoolFor<TModel>(
+            this IHtmlHelper<TModel> htmlHelper,
+            Expression<Func<TModel, bool>> propertyLambdaExpression,
+            LabelViewModel label,
```
Comment (DanCorder) [inline GovUkDesignSystem/GovUkHtmlHelperExtensions.cs]: Sorry, only just saw, this should have a default null value too

File: GovUkDesignSystem/HtmlGenerators/CheckboxFromBoolHtmlGenerator.cs [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+            string propertyId = idPrefix + htmlHelper.IdFor(propertyExpression);
+            string propertyName = idPrefix + htmlHelper.NameFor(propertyExpression);
+            htmlHelper.ViewData.ModelState.TryGetValue(propertyName, out var modelStateEntry);
+
+            // Get the value to put in the input from the post data if possible, otherwise use the value in the model 
+            var boolValue =
+                HtmlGenerationHelpers.GetBoolValueFromModelStateOrModel(htmlHelper.ViewData.Model,
+                    propertyExpression, modelStateEntry);
+
+            string classes = classOption;
```
Comment (DanCorder) [inline GovUkDesignSystem/HtmlGenerators/CheckboxFromBoolHtmlGenerator.cs]: This local variable is a bit pointless

**Review by DanCorder — CHANGES_REQUESTED** [review]

File: GovUkDesignSystem/GovUkHtmlHelperExtensions.cs (hunk 4 above)
Comment (asprouc) [reply GovUkDesignSystem/GovUkHtmlHelperExtensions.cs]: Done

File: GovUkDesignSystem/HtmlGenerators/CheckboxFromBoolHtmlGenerator.cs (hunk 5 above)
Comment (asprouc) [reply GovUkDesignSystem/HtmlGenerators/CheckboxFromBoolHtmlGenerator.cs]: Done

**Review by DanCorder — APPROVED** [review]

**Pull request merged**